packages/insights/src/insights-task-runner.ts

```diff
@@ -20,8 +26,55 @@ interface InsightsTaskRunnerOptions extends DefaultTasksRunnerOptions {
 type Context = {
   projectGraph: ProjectGraph;
   target: string;
+  nxJson: NxJson;
 };
 
+class InsightsRemoteCache implements RemoteCache {
+  constructor(private readonly axiosInstance: any) {}
+
+  async retrieve(hash: string, cacheDirectory: string): Promise<boolean> {
+    try {
+      const resp = await this.axiosInstance({
+        method: 'get',
+        url: `/nx-cache/${hash}`,
+        maxContentLength: 1000 * 1000 * 50
+      });
+      const tgz = path.join(cacheDirectory, `${hash}.tgz`);
+      writeFileSync(tgz, resp.data);
+      await tar.x({
+        file: tgz,
+        cwd: cacheDirectory
+      });
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  async store(hash: string, cacheDirectory: string): Promise<boolean> {
+    const tgz = path.join(cacheDirectory, `${hash}.tgz`);
+    try {
+      await tar.c(
+        {
+          gzip: false,
+          file: tgz,
+          cwd: cacheDirectory
+        },
+        [hash]
+      );
+      await this.axiosInstance({
+        method: 'post',
+        url: `/nx-cache/${hash}`,
+        data: { tgz: fs.readFileSync(tgz).toString() },
```

Comment:
Should be able to use a stream here so we don't need to read the file into memory fully.